packages/flutter/lib/src/painting/image_stream.dart

```diff
@@ -489,6 +491,13 @@ abstract class ImageStreamCompleter with Diagnosticable {
   /// if all [keepAlive] handles get disposed.
   bool _hadAtLeastOneListener = false;
 
+  /// Whether the listeners were added before a completer was attached
+  ///
+  /// This ultimately controls the synchronousCall parameter on the listeners.
+  /// When adding cached listeners after a completer was attached, [_hasInitialListeners] can 
+  /// be set to false to indicate to the listeners that they are being called asynchronously.
+  bool _hasInitialListeners = true;
```

Comment:
`has` feels a bit weird, only the ImageStream 'has' the initialize listener. `_addingInitialListeners` feels more self-explanatory.